applications/ContactStructuralMechanicsApplication/custom_linear_solvers/mixedulm_linear_solver.h

```diff
@@ -79,6 +79,12 @@ class MixedULMLinearSolver :
     ///@name Type Definitions
     ///@{
 
+    /// The flag that indicates if the blocks are allocated
+    KRATOS_DEFINE_LOCAL_FLAG( BLOCKS_ARE_ALLOCATED );
```

Comment:
why this change?

Reply:
because In order to reduce code duplication I have taken several methods from mixedulm_linear_solver.h  and moved to sparse_multiplication_utility. I changed also one member variable to a flag because in my opinion is cleaner